### DISCUSSION TITLE
Easiest way to specify correct data types for tabular?

### DISCUSSION THREAD

**Discussion by agt64** [autogluon/autogluon#1340] [Q&A] [answered]
I have a column that is a an integer code that is a label for a category. Autogluon thinks its an integer (very understandable) and I want to make sure it thinks it's a "category" or "object".  Unfortunately I'm having some difficulty properly specifying that. I created a dictionary of values that has the column name, and the data type it's supposed to be (column_types_dict_meta_cat). Then I tried:

feature_generator = AutoMLPipelineFeatureGenerator()
feature_generator.fit(train_data.drop(columns=[label]))

#get a dictionary of values that it inferred
feature_metadata_dict = feature_generator.feature_metadata_in.to_dict()

#change the dictionary to reflect proper types
for key, value in column_types_dict_meta_cat.items():
    feature_metadata_dict[key] = value

#create new feature Meta data with right types
feature_generator.feature_metadata_in = FeatureMetadata(type_map_raw = feature_metadata_dict) 
feature_generator.feature_metadata_in = feature_generator.feature_metadata_in.add_special_types(column_types_dict_list)

Then I added the feature generator to the fit function like this:
predictor = TabPredictor.fit(train_data, presets = presets, time_limit=time_limit, feature_generator=feature_generator)

The original data types are now correct, but the processed data types are not (still has integer data types). How can I correct this?

Thank you all for the help! Much appreciated.

**Comment by willsmithorg** [comment, marked as answer]
Quick workaround : convert it to a string and include some obviously string component into it to prevent it being converted back to an integer.

df['_column'] =  'x' + data['column'].astype(str)

Agreed, the API to help AutoGluon to coerce columns to certain datatypes could do with either better documentation, or clearer design, or both.  I'm thinking of contributing to the documentation i this area.

**Comment by Innixma** [comment]
Thanks both @magnate64 and @willsmithorg for the comments!

I've created #1437 to track the improvement of both documentation and override capabilities in this area, as currently it isn't very friendly to the user when they need to do something different from the default feature type inference logic.

For the case mentioned, @willsmithorg's answer is a good work-around until a better solution is provided in v0.4 release.
